pmultiqc/modules/quantms/histogram.py

```diff
@@ -0,0 +1,119 @@
+from collections import OrderedDict, defaultdict
+
+class Histogram:
+
+    def classify(self, plot_cat, with_remains = False, data_cats = None):
+        # with_remains -> Whether there are any remaining values outside the statistical range.
+        # data_cats -> Make statistics and display according to different data categories.
+        # plot_cat = 'frequency' -> Count the number of occurrences of different values.
+        # plot_cat = 'range' -> Count the frequency with which a value occurs over several consecutive ranges.
+        # plot_cat = 'record' -> Assigns values in order according to the different ranges provided.
+
+        self.data = dict()
```

Comment:
If you have data_cats separately, please use a list or a set.
If they have the same length, the correspondence is clear and the code will be faster.

Reply:
It also makes your life easier since you dont need to create the special name for the last bin always.